packages/cli/test/utils/simulation/assertions.ts

```diff
@@ -0,0 +1,307 @@
+  describe("operations", () => {
+    for (let n = 0; n < env.params.beaconNodes; n++) {
+      describe(`beacon node "${n}"`, () => {
+        describe("deposits", () => {
+          it("should process deposits in the block");
+        });
+
+        describe("graffiti", () => {
+          it("should process graffiti in the block");
+        });
+
+        describe("validator activation", () => {
+          it("should activates deposited validators");
+        });
+
+        describe("voluntary exist", () => {
+          it("should exists the validators with voluntary exits");
+        });
+      });
+    }
+  });
+}
+
+export function peersAssertions(env: SimulationEnvironment): void {
+  describe("peers", () => {
+    for (let n = 0; n < env.params.beaconNodes; n++) {
+      describe(`beacon node "${n}"`, () => {
+        it("should have correct number of peers connected");
+        describe("peer", () => {
+          it("should have gossip score >=0");
+          it("should have behavior penalty <= 0");
+          it("should have block provider score >= 0");
+          it("should have over all score >= 0");
+          it("should not have any validation error");
+        });
+      });
+    }
+  });
+}
+
+export function slashingAssertions(env: SimulationEnvironment): void {
+  describe("slashing", () => {
+    for (let n = 0; n < env.params.beaconNodes; n++) {
+      describe(`beacon node "${n}"`, () => {
+        describe("broadcast double vote to detect slasher", () => {
+          it("should slash expected number of validator");
+          it("should make a validator loose balance");
+        });
+
+        describe("propose double block to detect slasher", () => {
+          it("should slash expected number of validator");
+          it("should make a validator loose balance");
+        });
+      });
+    }
+  });
+}
+
+export function participationAssertions(env: SimulationEnvironment): void {
+  describe("participation", () => {
+    it("should have correct participation on head", async () => {
+      const states = await Promise.all(env.nodes.map((n) => n.api.debug.getStateV2("head")));
+
+      for (let n = 0; n < env.params.beaconNodes; n++) {
+        // As its end of epoch the "currentEpochParticipation" is all set to zero
+        const currentEpochParticipation = (states[n].data as altair.BeaconState).currentEpochParticipation;
+        // Make sure the its not end of epoch
+        expect(currentEpochParticipation.every((p) => p === 0)).to.be.eql(
+          true,
+          `Node ${n} has not set current epoch participation to zero. Probably its not the end of epoch.`
+        );
+
+        const previousEpochParticipation = (states[n].data as altair.BeaconState).previousEpochParticipation;
+        const totalAttestingBalance = previousEpochParticipation
+          .map((p, index) => ((p & TIMELY_HEAD) !== 0 ? states[n].data.balances[index] : 0))
+          .reduce((a, b) => a + b, 0);
```

Comment:
Write helper functions to compute participation. And consider just using a regular for loop would take one loop less over the data